### DISCUSSION TITLE
How to get list of PRIVATE repositories with api call

### DISCUSSION THREAD

**Discussion by Pcamachoc** [community/community#24787] [Apps, API and Webhooks]
<p>We are trying to access a user’s repositories list (including public and private). For this we have created a GitHub App in a github account and we have given read permissions to everything. We try to make the following calls by passing it the Oauth 2.0 token:</p>
<p><a href="https://api.github.com/users/USERNAME/repos" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/users/USERNAME/repos</a></p>
<p><a href="https://api.github.com/search/repositories?q=user:USERNAME" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/search/repositories?q=user:USERNAME</a></p>
<p>This only returns the user’s public repositories.</p>
<p>The next call also cannot access the private repositories<br>
<a href="https://api.github.com/repos/USERNAME/REPONAME/commits" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/repos/USERNAME/REPONAME/commits</a></p>
<p>We have also tried with {type: all} and {visibility: all}</p>
<p>Hope you can help me access all repositories and commits, thanks.</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/pcamachoc">@Pcamachoc</a>, any APIs will only provide publicly available information about another users GitHub account. Another users Private repository, or private email is not publicly available information.</p>
<p>You can list the the private repositories of the authenticated/current user, or repositories they have access to.<br>
Maybe something like <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-repositories-for-the-authenticated-user" rel="noopener nofollow ugc">List repositories for the authenticated user</a></p>

**Reply by ZainMughal1** [reply]
i'm following same docs but it show only public repo

**Comment by baselrabia** [comment]
<p><a class="mention" href="/byrneh">@byrneh</a>  I’m following the guide in the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-repositories-for-the-authenticated-user" rel="noopener nofollow ugc">List repositories for the authenticated user </a><br>
but I had private repos on the auth user account and I’m still getting empty array why is that happen</p>
<p>I’m listing user private repos throw /user/repos in GET request with the auth token<br>
I send parameter visibility in query with private value and it returns an empty array<br>
so can you help with that?</p>
<p><a href="https://api.github.com/user/repos?visibility=private" class="onebox" target="_blank" rel="noopener nofollow ugc">https://api.github.com/user/repos?visibility=private</a></p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="baselrabia" data-post="3" data-topic="136149">
<div class="title">
<div class="quote-controls"></div>
 baselrabia:</div>
<blockquote>
<p><a class="mention" href="/byrneh">@byrneh</a> I’m following the guide in the <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#list-repositories-for-the-authenticated-user" rel="noopener nofollow ugc">List repositories for the authenticated user </a><br>
but I had private repos on the auth user account and I’m still getting empty array why is that happen</p>
</blockquote>
</aside>
<p>I responded to your seperate thread on this</p><aside class="quote" data-post="2" data-topic="170842">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/cant-access-to-private-repos-with-api-call/170842/2">Can't Access to Private Repos with API Call</a> <a class="badge-wrapper  bullet" href="/c/github-ecosystem/github-rest-api/60"><span class="badge-category-parent-bg" style="background-color: #28a745;"></span><span class="badge-category-bg" style="background-color: #dcffe4;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="This is where you discuss our REST API. Further reading is found here:">REST API</span></a>
  </div>
  <blockquote>
    <a class="mention" href="/baselrabia">@baselrabia</a>, 
If your token is valid, in most cases a case of no data being returned or 404 not found from REST APIs is due to insufficient token scope permissions. 
You can view and edit you tokens permissions from this URL 
<a href="https://github.com/settings/tokens" rel="noopener nofollow ugc">https://github.com/settings/tokens</a> 
Does you token have repo scope permission when you tried <a href="https://docs.github.com/en/rest/reference/repos#list-repositories-for-the-authenticated-user" rel="noopener nofollow ugc">list-repositories-for-the-authenticated-user</a>?
  </blockquote>
</aside>
